http_graceful_test.go

```
@@ -125,5 +125,35 @@ func TestGraceful(t *testing.T) {
+		server := WithDefaults(&http.Server{
+			Addr:    "localhost:54935",
+			Handler: &testServer{delay: 0},
+		})
+
+		done := make(chan error)
+		go func() {
+			done <- Graceful(server.ListenAndServe, server.Shutdown)
+		}()
+
+		require.Eventually(t, func() bool {
+			res, err := http.Get("http://localhost:54935/")
```

golangci-lint (2.13.2)

[error] net/http.Get must not be called. use net/http.NewRequestWithContext and (*net/http.Client).Do(*http.Request)

(noctx)